### PR TITLE
Don't close CMIS Session & Bindings before reading the stream

### PR DESCRIPTION
**Pull request by tnleeuw**
No tests yet, need to think about how to make a good test for this.

### CODE REVIEW

**Review by jkosternl — APPROVED** [review]

File: cmis/src/main/java/nl/nn/adapterframework/extensions/cmis/CmisSessionWrapper.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+import org.apache.chemistry.opencmis.client.api.Session;
+import org.apache.chemistry.opencmis.commons.SessionParameter;
+import org.apache.chemistry.opencmis.commons.spi.CmisBinding;
+
+import lombok.Getter;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+public class CmisSessionWrapper implements AutoCloseable {
```
Comment (nielsm5) [inline cmis/src/main/java/nl/nn/adapterframework/extensions/cmis/CmisSessionWrapper.java:26]: Niet extends `SessionImpl`?

Comment (tnleeuw) [reply cmis/src/main/java/nl/nn/adapterframework/extensions/cmis/CmisSessionWrapper.java:26]: Dan moeten we ook de `SessionFactory` extenden, want die creeert de instance:

```
		SessionFactory sessionFactory = SessionFactoryImpl.newInstance();
		Session session = sessionFactory.createSession(parameterMap);
		log.debug("connected with repository [{}]", ()->getRepositoryInfo(session));

		return new CmisSessionWrapper(session);
```

Comment (nielsm5) [reply cmis/src/main/java/nl/nn/adapterframework/extensions/cmis/CmisSessionWrapper.java:26]: Ja klopt, die doet new SessionImpl(parameters, null, null, null, null)

**Comment by Laurens-makel** [comment]
Just wondering and trying to learn a bit here, the initial solution with a private inner class which takes the `Session` as constructor argument, is only used when runtime sessions are used.

What is the advantage of **always** wrapping the `Session` object created instead? It doesn't have to be an `AutoClosable` for global sessions right? Only when we want to schedule the close on pipeline exit..

**Comment by nielsm5** [comment]
> Just wondering and trying to learn a bit here, the initial solution with a private inner class which takes the `Session` as constructor argument, is only used when runtime sessions are used.
> 
> What is the advantage of **_always_** wrapping the `Session` object created instead? It doesn't have to be an `AutoClosable` for global sessions right? Only when we want to schedule the close on pipeline exit..

I think in this instance it's easier and makes the code more readable to always wrap the session.

**Review by nielsm5 — APPROVED** [review]

File: cmis/src/main/java/nl/nn/adapterframework/extensions/cmis/CmisSessionWrapper.java (hunk 1 above)
Comment (tnleeuw) [reply cmis/src/main/java/nl/nn/adapterframework/extensions/cmis/CmisSessionWrapper.java:26]: Ik zal dat doen wanneer ik een forward-port doe van de fix naar 7.9/8.0/master.
Voor 7.8 wil ik het zo laten als het nu is omdat het nu getest is zo door NN.

**Comment by tnleeuw** [comment]
> > > Just wondering and trying to learn a bit here, the initial solution with a private inner class which takes the `Session` as constructor argument, is only used when runtime sessions are used.
> > > What is the advantage of **_always_** wrapping the `Session` object created instead? It doesn't have to be an `AutoClosable` for global sessions right? Only when we want to schedule the close on pipeline exit..
> > 
> > 
> > I think in this instance it's easier and makes the code more readable to always wrap the session.
> 
> In my opinion, these changes makes the code actually less readable and could be avoided.
> 

I think it makes some parts more readable, and others get a bit an extra call.

For 7.9 and forward I will try to extend the session / sessionfactory instead of wrapping it, as Niels suggested, and that would address all concerns regarding code quality and consistency.

**Pull request merged**